packages/core/src/components/AutocompleteField/AutocompleteField.jsx

```diff
@@ -0,0 +1,198 @@
+                  <h5
+                    className="ds-u-margin--0 ds-u-padding--1"
+                    id="constrained-list-header"
+                  >
+                    {constrainedListText}
+                  </h5>
+                ) : null}
+
+                <ul
+                  aria-labelledby={
+                    constrainedList ? 'constrained-list-header' : null
+                  }
+                  className="ds-c-list--bare"
+                  role="listbox"
+                >
+                  {items
+                    .filter(
+                      item =>
+                        !inputValue ||
+                        item.name
+                          .toLowerCase()
+                          .includes(inputValue.toLowerCase())
+                    )
+                    .map((item, index) => (
+                      <li
+                        aria-selected={highlightedIndex === index}
+                        className={
+                          highlightedIndex === index
+                            ? 'ds-u-padding--1 ds-c-autocomplete__list-item ds-c-autocomplete__list-item--active'
+                            : 'ds-u-padding--1 ds-c-autocomplete__list-item'
+                        }
+                        key={item.id}
+                        role="option"
+                        {...getItemProps({ item })}
+                      >
+                        {itemToString(item)}
+                      </li>
+                    ))}
+                </ul>
+              </div>
+            ) : null}
+
+            <Button
+              aria-label={clearAriaLabel}
+              className="ds-u-float--right ds-u-padding-right--0 ds-c-autocomplete__button"
+              href="javascript:void(0);"
+              onClick={clearSelection}
+              size="small"
+              variation="transparent"
+            >
+              {clearInputText}
+            </Button>
+          </div>
+        )}
+      />
+    );
+  }
+}
+
+AutocompleteField.defaultProps = {
+  clearAriaLabel: 'Clear typeahead and search again',
+  clearInputText: 'Search again',
+  constrainedListText: 'Select from the options below:'
+};
+
+AutocompleteField.propTypes = {
+  children: PropTypes.node,
+  /**
+   * Screenreader-specific label for the Clear input link. Intended to provide a longer, more descriptive explanation of the link's behavior.
+   */
+  clearAriaLabel: PropTypes.string,
+  /**
+   * Clear link text that will appear on the page as part of the rendered component
+   */
+  clearInputText: PropTypes.string,
+  /**
+   * Adds a conditional header to `<div.ds-c-autocomplete__list>`
+   */
+  constrainedList: PropTypes.bool,
+  /**
+   * Text string for the `<h5>` header added to the`<div.ds-c-autocomplete__list>` when prop `constrainedList`is passed to `<Autocomplete>`
+   */
+  constrainedListText: PropTypes.string,
+  /**
+   * Passes prop `disabled` to the child `<Textfield>` component
```

Comment:
This is a little confusing to me. Could we remove the need for this prop at this level, or rephrase the documentation around it (ie. "Disables autocomplete functionality and overrides the `TextField`'s disabled state")? 

I could be wrong, but I think the use case of having a disabled autocomplete field is low, and if someone wanted that, they could render `<Textfield disabled />` (without nesting it in `Autocomplete`)

Reply:
Done. Removed the `disabled` prop. If we hear from users that having a way to disable the Clear search link is needed, we can refactor at that time.